client/providers/mocks/oidp_server.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 OpenPubkey
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package mocks
+
+import (
+	"crypto"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/zitadel/oidc/v2/pkg/oidc"
+)
+
+const (
+	jwksEndpoint            = "/.well-known/jwks.json"
+	wellKnownConfigEndpoint = "/.well-known/openid-configuration"
+)
+
+type OIDPServer struct {
```

Comment:
This looks like the JWKS Server from my MFA example code.  Name seems a little confusing because it just serves well-known configuration and JWKS isn't an OP. Maybe call it `JwksServer` or `JWKSServer` unless the plan is to add more OP functionality to it.

Not a blocker